mackerel-plugin-php-fpm/lib/php-fpm.go

```diff
@@ -17,6 +21,91 @@ type PhpFpmPlugin struct {
 	Prefix      string
 	LabelPrefix string
 	Timeout     uint
+	Socket      SocketFlag
+}
+
+// SocketFlag represents -socket flag.
+type SocketFlag struct {
+	u       *url.URL
+	Network string
+	Address string
+}
+
+func (p *SocketFlag) String() string {
+	if p.u == nil {
+		return ""
+	}
+	return p.u.String()
+}
+
+// Set implements flag.Value interface.
+func (p *SocketFlag) Set(s string) error {
+	*p = SocketFlag{}
+	u, err := parseSocketFlag(s)
+	if err != nil {
+		return err
+	}
+	switch u.Scheme {
+	case "tcp":
+		p.Network = "tcp"
+		p.Address = u.Host
+	case "unix":
+		p.Network = "unix"
+		p.Address = u.Path
+	default:
+		panic("unknown scheme: " + u.Scheme)
```

Comment:
I think returning error is better than using panic.

Reply:
fixed.